paderbox/utils/nested.py

```diff
@@ -400,3 +413,295 @@ def squeeze_nested(orig):
         if squeezed and all([orig[key] == orig[keys[0]] for key in keys]):
             return orig[keys[0]]
     return orig
+
+
+def get_by_path(
+        container: Union[Mapping, Sequence],
+        path: Union[str, Tuple[Any, ...], None],
+        *,
+        allow_early_stopping: bool = False,
+        sep: str = '.',
+        default: Any = ...,
```

Comment:
Why do you use `Ellipsis` here?
I think you wanted to use something like:

```
NO_VALUE = object()
def get_by_path(..., default=NO_VALUE): ...
```

Reply:
Yes, I saw the `Ellipsis` being used for this somewhere (I don't remember where), but your suggestion is better in case anyone want's to have `Ellipsis` as default. Should we define a global `paderbox.NO_VALUE` constant?

Reply:
Ups, I forgot the `_` in `_NO_VALUE`.

I think it is common to define the `NO_VALUE` as a hidden variable, in this way it is ensured, that no one can use it wrong. I checked `inspect` and the have only `inspect.Parameter.empty` as public (i.e. `inspect._empty` and `inspect._void` are private)

Reply:
I just realized that the private `_NO_VALUE` constant makes it hard to wrap this function without accessing the private module member, so it might be better to have it public.